addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java

```diff
@@ -51,16 +54,22 @@ public class CrossDomainScriptInclusionScanRule extends PluginPassiveScanner {
             LogManager.getLogger(CrossDomainScriptInclusionScanRule.class);
     private Model model = null;
 
+    private final TrustedDomains trustedDomains = new TrustedDomains();
+
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        trustedDomains.update(getConfig().getString(RuleConfigParam.RULE_DOMAINS_TRUSTED, ""));
+        trustedDomains.add(new RegexTrust(msg.getRequestHeader().getHostName()));
+
         if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isHtml()) {
             List<Element> sourceElements = source.getAllElements(HTMLElementName.SCRIPT);
             if (sourceElements != null) {
                 for (Element sourceElement : sourceElements) {
                     String src = sourceElement.getAttributeValue("src");
                     if (src != null
-                            && isScriptFromOtherDomain(
-                                    msg.getRequestHeader().getHostName(), src, msg)) {
+                                    && isScriptFromOtherDomain(
+                                            msg.getRequestHeader().getHostName(), src, msg)
+                            || !trustedDomains.isIncluded(msg.getRequestHeader().getHostName())) {
```

Comment:
`&&` would be more clear.

Reply:
Versus "or not".

Reply:
The negation should be kept even when changed to `&&` (that's why it's clearer using the and). Note also that it should check the domain of the `src` attribute not the domain of the message.